Add integration testing

daninfpj opened the pull request:
This PR adds integrations testing  (#2). It loads the example app as a submodule and runs the tests on top of that.

The example app was modified to remove CSS and JS assets because they were causing trouble when running the tests.


There's [a couple steps](https://github.com/daninfpj/webpacker-react/tree/testing#testing) that should be run before running the tests but I’m not sure if we should add them to the rake task.

There's also a couple additional commits:

- Rubocop was added to check style issues and the rules were copied over from rails
- The other one changes the helper we were using to render the div tag, because it was generating self-closing tags that eat the content that comes after them (similar to this [issue](https://github.com/rails/rails/issues/4919))

renchap commented on .rubocop.yml:
```diff
@@ -0,0 +1,120 @@
+  Exclude:
+    - '**/templates/**/*'
+    - '**/vendor/**/*'
+    - 'actionpack/lib/action_dispatch/journey/parser.rb'
```
This is probably not needed

daninfpj replied:
Fixed. Left only `'**/vendor/**/*'`as some node_modules include ruby files.

renchap commented on test/rails_helper.rb:
```diff
@@ -1,4 +1,12 @@
 require "rails/test_help"
+require "rails/generators"
+require "pathname"
+require 'minitest/mock'
```
Use double quotes for consistence

daninfpj replied:
Fixed.

daninfpj commented:
I added the submodule precisely because I wanted to avoid the duplication. In the end it was actually a bit cumbersome to work with it as you mention, so I removed it.

Also, I'm now using `yarn` everywhere.

renchap commented on test/example_app/README.md:
```diff
@@ -0,0 +1,55 @@
+# Webpacker-React sample application
```
Not needed

renchap commented on test/example_app/app/assets/config/manifest.js:
```diff
@@ -0,0 +1,3 @@
+//= link_tree ../images
```
Not needed. Might be a good idea to remove the whole ´app/assets' directory if it does not break the app

renchap commented on test/example_app/app/helpers/pages_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module PagesHelper
```
Same for helpers, lets try to keep the files down to a minimum
